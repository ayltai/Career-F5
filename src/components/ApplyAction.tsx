import { RocketLaunch as RocketLaunchIcon, } from '@mui/icons-material';
import { Button, Typography, } from '@mui/material';
import React from 'react';
import { useTranslation, } from 'react-i18next';

export const ApplyAction = ({
    onClick,
} : {
    onClick : () => void,
}) => {
    const { t, } = useTranslation();

    return (
        <Button
            color='primary'
            variant='contained'
            startIcon={
                <RocketLaunchIcon sx={{
                    color : 'white',
                }} />
            }
            onClick={onClick}>
            <Typography
                color='white'
                variant='button'>
                {t('action_apply')}
            </Typography>
        </Button>
    );
};
