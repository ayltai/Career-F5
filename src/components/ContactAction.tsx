import { WhatsApp as WhatsAppIcon, } from '@mui/icons-material';
import { Button, Typography, } from '@mui/material';
import React from 'react';
import { useTranslation, } from 'react-i18next';

const handleClick = () => {
    if (process.env.REACT_APP_WHATSAPP_PHONE_NUMBER) {
        const newWindow = window.open(`https://wa.me/${process.env.REACT_APP_WHATSAPP_PHONE_NUMBER}`, '_blank', 'noopener,noreferrer');
        if (newWindow) newWindow.opener = null;
    }
};

export const ContactAction = () => {
    const { t, } = useTranslation();

    return (
        <Button
            variant='contained'
            color='secondary'
            startIcon={
                <WhatsAppIcon sx={{
                    color : 'white',
                }} />
            }
            onClick={handleClick}>
            <Typography
                color='white'
                variant='button'>
                {t('whatsapp')}
            </Typography>
        </Button>
    );
};
