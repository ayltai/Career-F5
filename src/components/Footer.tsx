import { Box, Container, Typography, } from '@mui/material';
import React from 'react';
import { useTranslation, } from 'react-i18next';

export const Footer = () => {
    const { t, } = useTranslation();

    return (
        <Box
            sx={{
                backgroundColor: 'black',
            }}
            width='100%'>
            <Container maxWidth='lg'>
                <Box
                    paddingY={4}
                    textAlign='center'>
                    <Typography
                        color='white'
                        variant='overline'>
                        {t('copyright')}
                    </Typography>
                </Box>
            </Container>
        </Box>
    );
};
