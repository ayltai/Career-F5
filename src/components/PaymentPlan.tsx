import { ExpandMore as ExpandMoreIcon, } from '@mui/icons-material';
import { Accordion, AccordionSummary, AccordionDetails, Box, Typography, useMediaQuery, } from '@mui/material';
import React from 'react';

export const PaymentPlan = ({
    title,
    description,
    cost,
} : {
    title       : string,
    description : string,
    cost        : string,
}) => {
    const desktopMode = useMediaQuery('(min-width:900px)');

    return (
        <Accordion
            disableGutters
            square
            sx={{
                ':hover'      : {
                    boxShadow : 8,
                },
            }}
            elevation={2}>
            <AccordionSummary expandIcon={<ExpandMoreIcon />}>
                <Typography
                    width={desktopMode ? '50%' : '100%'}
                    flexShrink={desktopMode ? 0 : undefined}
                    variant="h5"
                    fontWeight='bold'>
                    {title}
                </Typography>
                {desktopMode && (
                    <Typography
                        color='secondary'
                        variant="h5"
                        fontWeight='bold'>
                        {cost}
                    </Typography>
                )}
            </AccordionSummary>
            <AccordionDetails>
                {!desktopMode && (
                    <Box paddingY={2}>
                        <Typography
                            color='secondary'
                            variant="h5"
                            fontWeight='bold'>
                            {cost}
                        </Typography>
                    </Box>
                )}
                <Typography>
                    {description}
                </Typography>
            </AccordionDetails>
        </Accordion>
    );
};
