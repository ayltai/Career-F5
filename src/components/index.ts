export { ApplyAction, } from './ApplyAction';
export { ApplyDialog, } from './ApplyDialog';
export { ContactAction, } from './ContactAction';
export { Footer, } from './Footer';
export { PaymentPlan, } from './PaymentPlan';
export { RangeIndicator, } from './RangeIndicator';
export { TopBar, } from './TopBar';
