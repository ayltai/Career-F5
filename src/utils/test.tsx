import { ThemeProvider, } from '@mui/material';
import { configureStore, PreloadedState, Store, } from '@reduxjs/toolkit';
import '@testing-library/jest-dom/extend-expect';
import { render, } from '@testing-library/react';
import mediaQuery from 'css-mediaquery';
import React, { ReactNode, } from 'react';
import { Provider, } from 'react-redux';

import { preferencesReducer, } from '../states';
import { AppTheme, } from '../styles';

export const createMatchMedia = (width : number) => (query : string) => ({
    matches             : mediaQuery.match(query, {
        width,
    }),
    media               : '',
    onchange            : null,
    addListener         : () => {},
    removeListener      : () => {},
    addEventListener    : () => {},
    removeEventListener : () => {},
    dispatchEvent       : () => false,
});

const createStore = (preloadedState? : Record<string, any>) => configureStore({
    reducer : {
        preferences : preferencesReducer,
    },
    preloadedState,
});

const defaultStore = createStore();

type RootState = ReturnType<typeof defaultStore.getState>;

const customRender = (ui : any, {
    preloadedState,
    store = createStore(preloadedState),
    ...rest
} : {
    preloadedState?  : PreloadedState<RootState>,
    store?           : Store<RootState>,
    [ rest : string] : any,
} = {}) => render(ui, {
    wrapper : ({
        children,
    } : {
        children : ReactNode,
    }) => (
        <Provider store={store}>
            <ThemeProvider theme={AppTheme('en')}>
                {children}
            </ThemeProvider>
        </Provider>
    ),
    ...rest,
});

export * from '@testing-library/react';

export { customRender as render, };
