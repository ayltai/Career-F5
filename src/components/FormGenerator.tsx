import { Button, Card, CardActions, CardContent, Typography, useMediaQuery, } from '@mui/material';
import { GridContainer, widgets, } from '@ui-schema/ds-material';
import { createEmptyStore, createOrderedMap, injectPluginStack, isInvalid, relT, StoreSchemaType, storeUpdater, TranslatorContext, UIMetaProvider, UIStoreProvider, } from '@ui-schema/ui-schema';
import React, { useCallback, useState, } from 'react';
import { useTranslation, } from 'react-i18next';

import { handleError, } from '../utils';

const GridStack = injectPluginStack(GridContainer);

export type FormGeneratorProps = {
    title?       : string,
    description? : string,
    definition   : object,
    onSubmit     : (data : Record<string, string>) => Promise<boolean>,
};

export const FormGenerator = ({
    elevated = true,
    title,
    description,
    definition,
    onSubmit,
} : FormGeneratorProps & {
    elevated? : boolean,
}) => {
    const storeSchema = createOrderedMap({
        type : 'object',
        ...definition,
    });

    const [ store,        setStore,        ] = useState(() => createEmptyStore(storeSchema.get('type')));
    const [ showValidity, setShowValidity, ] = useState(false);
    const [ isSubmitting, setIsSubmitting, ] = useState(false);

    const desktopMode = useMediaQuery('(min-width:900px)');

    const { t, } = useTranslation();

    const handleChange = useCallback((actions : any) => setStore(storeUpdater(actions)), [ setStore,]);

    const handleSubmit = () => {
        setShowValidity(true);

        if (!isInvalid(store.getValidity())) {
            setIsSubmitting(true);

            onSubmit(store.valuesToJS() as Record<string, string>).then(success => {
                if (success) {
                    setStore(createEmptyStore(storeSchema.get('type')));

                    setShowValidity(false);
                }
            }).then(() => setIsSubmitting(false)).catch(handleError);
        }
    };

    const translate = (text : string, context? : TranslatorContext, schema? : StoreSchemaType) => {
        const schemaT = relT(schema, context);

        return schemaT ? schemaT : t(text);
    };

    return (
        <Card
            sx={{
                borderRadius : 4,
                padding      : desktopMode ? 4 : 1,
            }}
            elevation={elevated ? 4 : 0}>
            <CardContent>
                {title && (
                    <Typography
                        gutterBottom
                        variant='h5'
                        fontWeight='bold'>
                        {t(title)}
                    </Typography>
                )}
                {description && (
                    <Typography
                        gutterBottom
                        variant='body1'>
                        {t(description)}
                    </Typography>
                )}
                <UIMetaProvider
                    t={translate}
                    widgets={widgets}>
                    <UIStoreProvider
                        showValidity={showValidity}
                        store={store}
                        onChange={handleChange}>
                        <GridStack
                            isRoot
                            schema={storeSchema} />
                    </UIStoreProvider>
                </UIMetaProvider>
            </CardContent>
            <CardActions>
                <Button
                    sx={{
                        marginX : 1,
                    }}
                    disabled={isSubmitting || !!isInvalid(store.getValidity())}
                    variant='contained'
                    onClick={handleSubmit}>
                    <Typography
                        color='white'
                        variant='button'>
                        {t(isSubmitting ? 'action_submitting' : 'action_submit')}
                    </Typography>
                </Button>
            </CardActions>
        </Card>
    );
};
