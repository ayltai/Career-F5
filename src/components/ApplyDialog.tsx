import { Close as CloseIcon, } from '@mui/icons-material';
import { Box, Dialog, DialogContent, DialogContentText, DialogTitle, IconButton, Typography, useMediaQuery, useTheme, } from '@mui/material';
import React from 'react';
import { useTranslation, } from 'react-i18next';

import { FormGenerator, FormGeneratorProps, } from './FormGenerator';

export const ApplyDialog = ({
    open = false,
    onClose,
    title,
    description,
    definition,
    onSubmit,
} : FormGeneratorProps & {
    open?    : boolean,
    onClose? : () => void,
}) => {
    const theme = useTheme();

    const desktopMode = useMediaQuery('(min-width:900px)');

    const { t, } = useTranslation();

    return (
        <Dialog
            open={open}
            onClose={onClose}>
            <DialogTitle>
                <Box
                    paddingX={desktopMode ? 4 : 1}
                    paddingTop={desktopMode ? 4 : 1}>
                    <Typography variant='h5'>
                        {title}
                    </Typography>
                    <IconButton
                        sx={{
                            position : 'absolute',
                            right    : desktopMode ? 16 : 8,
                            top      : desktopMode ? 16 : 8,
                            color    : theme.palette.text.secondary,
                        }}
                        onClick={onClose}>
                        <CloseIcon />
                    </IconButton>
                </Box>
            </DialogTitle>
            <DialogContent>
                <DialogContentText>
                    <Typography
                        gutterBottom
                        paddingX={desktopMode ? 8 : 4}
                        color='text.secondary'
                        variant='subtitle1'
                        fontStyle='italic'>
                        {t('apply_note')}
                    </Typography>
                    <Typography
                        gutterBottom
                        paddingX={desktopMode ? 4 : 1}
                        paddingTop={2}
                        color='text.primary'
                        variant='body1'>
                        {description}
                    </Typography>
                </DialogContentText>
                <FormGenerator
                    elevated={false}
                    definition={definition}
                    onSubmit={onSubmit} />
            </DialogContent>
        </Dialog>
    );
};
