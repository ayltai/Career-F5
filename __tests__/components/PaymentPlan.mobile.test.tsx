import React from 'react';

import { createMatchMedia, render, } from '../../src/utils/test';
import { PaymentPlan, } from '../../src/components/PaymentPlan';

describe('PaymentPlan', () => {
    beforeAll(() => {
        window.matchMedia = createMatchMedia(360);
    });

    it('renders correctly for mobile', () => expect(render(
        <PaymentPlan
            title='title'
            description='description'
            cost='cost' />
    )).toMatchSnapshot());
});
