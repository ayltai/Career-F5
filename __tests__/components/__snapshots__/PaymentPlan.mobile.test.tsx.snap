// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentPlan renders correctly for mobile 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation2 MuiAccordion-root css-y7m5ga-MuiPaper-root-MuiAccordion-root"
      >
        <div
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root css-1pvvkxv-MuiButtonBase-root-MuiAccordionSummary-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-s668vz-MuiTypography-root"
            >
              title
            </h5>
          </div>
          <div
            class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ExpandMoreIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                class="MuiAccordion-region"
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                >
                  <div
                    class="MuiBox-root css-1sf3xto"
                  >
                    <h5
                      class="MuiTypography-root MuiTypography-h5 css-1uzjrf3-MuiTypography-root"
                    >
                      cost
                    </h5>
                  </div>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-1roozjh-MuiTypography-root"
                  >
                    description
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation2 MuiAccordion-root css-y7m5ga-MuiPaper-root-MuiAccordion-root"
    >
      <div
        aria-expanded="false"
        class="MuiButtonBase-root MuiAccordionSummary-root css-1pvvkxv-MuiButtonBase-root-MuiAccordionSummary-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-s668vz-MuiTypography-root"
          >
            title
          </h5>
        </div>
        <div
          class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ExpandMoreIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </div>
      </div>
      <div
        class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
          >
            <div
              class="MuiAccordion-region"
              role="region"
            >
              <div
                class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
              >
                <div
                  class="MuiBox-root css-1sf3xto"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 css-1uzjrf3-MuiTypography-root"
                  >
                    cost
                  </h5>
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1roozjh-MuiTypography-root"
                >
                  description
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
